client/client.go

```diff
@@ -31,23 +31,44 @@ type Client struct {
 	client    proto.DelegatedRouting_Client
 	validator record.Validator
 
-	provider *Provider
-	identity crypto.PrivKey
+	provider          *Provider
+	providerBatchSize int
+	identity          crypto.PrivKey
+}
+
+type Config struct {
+	Provider         *Provider
+	Identity         crypto.PrivKey
+	ProvideBatchSize int
+}
+
+// Validate validates the current config, and also sets defaults values when needed.
+func (c *Config) Validate() error {
+	if c.Provider != nil && !c.Provider.Peer.ID.MatchesPublicKey(c.Identity.GetPublic()) {
+		return errors.New("identity does not match provider")
+	}
+
+	if c.ProvideBatchSize == 0 {
+		c.ProvideBatchSize = 30000 // this will generate payloads of ~1MB in size
```

Comment:
>nit (1): this comes with various assumptions about average size of a single item, which will quickly get outdated when we have WebTransport enabled by default (/webtransport multiaddrs with two /certhash segments will baloon the size of the batch beyond initial estimate), or add more transports in the future.

Then we need to add a specific limit on the amount of multiaddrs allowed. But that is not the problem right now.

Checking the raw byte size of the payload will make the code hard to read and understand. In my opinion, is not worth it (have a payload of ~2Mb instead of ~900Kb because we added more multiaddresses)